src/utils/tree_node.C

```diff
@@ -485,40 +493,33 @@ const Elem* TreeNode<N>::find_element(const Point& p, const std::set<subdomain_i
 
 
 template <unsigned int N>
-const Elem* TreeNode<N>::find_element_in_children(const Point& p, const std::set<subdomain_id_type> *allowed_subdomains) const
+const Elem* TreeNode<N>::find_element_in_children
+(const Point& p,
+ const std::set<subdomain_id_type> *allowed_subdomains,
+ Real relative_tol) const
 {
   libmesh_assert (!this->active());
 
-  unsigned int excluded_child = libMesh::invalid_uint;
+  std::vector<bool> searched_child(children.size(), false);
```

Comment:
> Do you have any more details on what you encountered in MOOSE?  It wouldn't be the first time we'd had to work around a common compiler bug.

Looks like the [thing I was remembering](https://github.com/idaholab/moose/commit/579e54fd64cfa58b14ffe96c5ba61a828c593205) was doing non-const modifications...

Still, I don't see the point of introducing *even the possibility* of hard-to-diagnose future threading bugs by continuing to use `vector<bool>` in any capacity.  Then again, since everyone knows about the issues, maybe it isn't that hard to diagnose...